pkg/operator/target_config_reconciler_v311_00.go

```diff
@@ -50,6 +52,10 @@ func createTargetConfigReconciler_v311_00_to_latest(c TargetConfigReconciler, re
 		errors = append(errors, fmt.Errorf("%q: %v", "configmap/kube-apiserver-pod", err))
 	}
 
+	if _, err = managePrometheus_v311_00_to_latest(c.kubeClient.RbacV1(), c.dynamicClient, recorder); err != nil {
```

Comment:
model all this as a separate control loop.  This loop shouldn't trigger a failing status and is logically generic for all of our operators.

Reply:
i think it makes sense to make this generic. https://github.com/openshift/library-go/pull/150